test/functional/rpc_fundrawtransaction.py

```diff
@@ -103,9 +103,15 @@ def run_test(self):
         self.generate(self.nodes[2], 1)
         self.generate(self.nodes[0], 121)
 
+        # coins for simple tests
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.5)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.0)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 5.0)
+
+        self.generate(self.nodes[0], 1)
```

Comment:
Did you duplicate this `self.generate(self.nodes[0], 1)`, any reason? You moved it to here but you also added it in L684.

Reply:
Yes, generate mines a block so it has been duplicated to confirm the transactions preceding it.

Reply:
It's only a suggestion, but I think this way would fit better. 
```diff
diff --git a/test/functional/rpc_fundrawtransaction.py b/test/functional/rpc_fundrawtransaction.py
index 338c31eea..71fda6362 100755
--- a/test/functional/rpc_fundrawtransaction.py
+++ b/test/functional/rpc_fundrawtransaction.py
@@ -103,13 +103,7 @@ class RawTransactionsTest(BitcoinTestFramework):
         self.generate(self.nodes[2], 1)
         self.generate(self.nodes[0], 121)
 
-        # coins for simple tests
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.5)
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.0)
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 5.0)
-
-        self.generate(self.nodes[0], 1)
-
+        self.test_watchonly()
         self.test_change_position()
         self.test_simple()
         self.test_simple_two_outputs()
@@ -132,7 +126,6 @@ class RawTransactionsTest(BitcoinTestFramework):
         self.test_many_inputs_fee()
         self.test_many_inputs_send()
         self.test_op_return()
-        self.test_watchonly()
         self.test_all_watched_funds()
         self.test_option_feerate()
         self.test_address_reuse()
@@ -681,6 +674,11 @@ class RawTransactionsTest(BitcoinTestFramework):
         self.nodes[0].lockunspent(False, [{"txid": self.watchonly_txid, "vout": self.watchonly_vout}])
 
         self.nodes[0].sendtoaddress(self.nodes[3].get_wallet_rpc(self.default_wallet_name).getnewaddress(), self.watchonly_amount / 10)
+                # coins for simple tests
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.5)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.0)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 5.0)
+
         self.generate(self.nodes[0], 1)
 
         inputs = []
```